From 0a22588c5e68ea8a3da6310cca0f1f761227726a Mon Sep 17 00:00:00 2001
From: Finley Xiao <finley.xiao@rock-chips.com>
Date: Tue, 9 May 2017 21:53:37 +0800
Subject: [PATCH] PM / devfreq: rockchip_dmc: avoid waiting for vop line flag
 indefinitely

It may disable vop_crtc when scaling frequency, in this case,
devfreq thread will wait for vop line flag indefinitely, the
system will crash.

Change-Id: I7043b285c329ff23e2fc9c5b5f3a165c37ef6378
Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
---
 drivers/devfreq/rockchip_dmc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/devfreq/rockchip_dmc.c b/drivers/devfreq/rockchip_dmc.c
index 26ead87c138e..00cb8eba65c3 100644
--- a/drivers/devfreq/rockchip_dmc.c
+++ b/drivers/devfreq/rockchip_dmc.c
@@ -899,6 +899,9 @@ static int rockchip_dmcfreq_probe(struct platform_device *pdev)
 	data->dev = dev;
 	platform_set_drvdata(pdev, data);
 
+	if (rockchip_drm_register_notifier_to_dmc(data->devfreq))
+		dev_err(dev, "drm fail to register notifier to dmc\n");
+
 	if (rockchip_pm_register_notify_to_dmc(data->devfreq))
 		dev_err(dev, "pd fail to register notify to dmc\n");
 
-- 
2.35.3

